uaclient/defaults.py

```diff
@@ -12,6 +12,8 @@
 DEFAULT_PRIVATE_MACHINE_TOKEN_PATH = (
     DEFAULT_DATA_DIR + PRIVATE_SUBDIR + "/" + MACHINE_TOKEN_FILE
 )
+CANDIDATE_REGEX = r"Candidate: (?P<candidate>.*?)\n"
+CANDIDATE_CACHE_PATH = "/tmp/ua-candidate"
```

Comment:
I suggest we give ourselves a directory in `/run` because we'll likely want more temporary files in the future and we should avoid polluting the top level of /tmp or /run
```suggestion
CANDIDATE_CACHE_PATH = "/run/ubuntu-advantage/candidate-version"
```

Reply:
The problem is that non-root cannot create things in `/run`

Reply:
Ah you're right okay. How about we just create our own directory in tmp then? `/tmp/ubuntu-advantage/candidate-version`?

Reply:
should we go for `ubuntu-advantage` (package name) or `pro-client`?

Reply:
went for `ubuntu-advantage`, just as the `/etc` one